### DISCUSSION TITLE
Change session when branch changes

### DISCUSSION THREAD

**Discussion by GuillaumeLagrange** [olimorris/persisted.nvim#70] [Ideas]
Hello

First of all, thank you for this very efficient plugin.
My workflow includes this plugin with this config

```
autoload = true
autosave = true
use_git_branch = true
```

It works almost perfectly, with the small inconvenience that when I check out a new branch, the "currently opened session" is still the session of the old branch.

I have looked into the source code and started thinking about how I would implement session changing (either automatically or by proposing it to the user with a dialog), and have a few ideas. I was looking for a small PR to start honing my lua skills and I think this is it.

I am making this post to make sure the feature is something that looks reasonable to other people than myself 😅 
We could also include a config option to enable this feature and have it disabled by default.

Thanks in advance

**Comment by olimorris** [comment]
I think this would be a great idea and I'd be totally open to the idea. I'm not sure how much of this feature would reside within the plugin though. Assuming you execute a Lua function to switch branches, you could call `SessionLoad` or `require("persisted").load()`.

I've tried to keep this plugin as simple as possible so generally not keen on any user dialogs.

**Reply by GuillaumeLagrange** [reply]
Thank you for the fast response. I usually checkout from another terminal, so calling the load function would not really fix my problem.

I'll try to come up with something clever and open a PR which keeps the plugin as simple as it already is.

Worst case scenario I will just come up with a user callback that does what I want.
